packages/core/src/snapshot.js

```diff
@@ -361,6 +362,11 @@ export function createSnapshotsQueue(percy) {
     .handle('task', async function*({ resources, ...snapshot }) {
       let { name, meta } = snapshot;
 
+      if (percy.client.screenshotFlow === 'automate' && percy.client.buildType !== 'automate') {
+        throw new Error(`Cannot run automate screenshots in ${percy.client.buildType} project. Please use automate project token`);
+      } else if (percy.client.screenshotFlow === 'app' && percy.client.buildType !== 'app') {
+        throw new Error(`Cannot run App Percy screenshots in ${percy.client.buildType} project. Please use App Percy project token`);
```

Comment:
We can simplify this to below
```suggestion
     const errorMessage = percy.client.screenshotFlow === 'automate' ? 'automate' : 'App Percy';
const projectType = percy.client.screenshotFlow === 'automate' ? 'automate' : 'app';

if (percy.client.buildType !== projectType) {
  throw new Error(`Cannot run ${errorMessage} screenshots in ${percy.client.buildType} project. Please use ${errorMessage} project token`);
}
```

Reply:
Did not do that reason being web projects also goes through this flow, and didn't wanted to touch web projects as it can have two flows snapshot and exec and will increase the blast radius and we already have a validation for it on sdk side and api side.

Reply:
Ack.